engine/src/mac-core.mm

```diff
@@ -1949,49 +1937,51 @@ MCPlatformModifiers MCMacPlatformMapNSModifiersToModifiers(NSUInteger p_modifier
 
 ////////////////////////////////////////////////////////////////////////////////
 
+void get_desktop_height()
+{
+    s_desktop_height = 0.0f;
+    
+    for (NSScreen * t_screen in [NSScreen screens])
+    {
+        NSRect t_rect = [t_screen frame];
+        if (t_rect.origin.y + t_rect.size.height > s_desktop_height)
+            s_desktop_height = t_rect.origin.y + t_rect.size.height;
+    }
+    
+    s_have_desktop_height = true;
+}
+
 void MCMacPlatformMapScreenMCPointToNSPoint(MCPoint p, NSPoint& r_point)
 {
-	if (!s_have_primary_screen_height)
-	{
-		MCRectangle t_viewport;
-		MCPlatformGetScreenViewport(0, t_viewport);
-	}
+	if (!s_have_desktop_height)
+        get_desktop_height();
```

Comment:
I suggest you make `get_desktop_height()` return the desktop height, and make that the only function that accesses the static variables.  Then, for example:

```c++
void MCMacPlatfromMapScreenMCPointToNSPoint(MCPoint p, NSPoint &r_point)
{
    r_point = NSMakePoint(p.x, get_desktop_height() - p.y);
}
```